lib/elixir/lib/file/stream.ex

```diff
@@ -116,8 +128,11 @@ defmodule File.Stream do
           {:error, __MODULE__}
 
         {:ok, %{size: size}} ->
+          bom_offset = count_raw_bom(stream, modes)
+          offset = get_read_offset(modes)
+          size = max(size - bom_offset - offset, 0)
           remainder = if rem(size, bytes) == 0, do: 0, else: 1
-          {:ok, div(size, bytes) + remainder - count_raw_bom(stream, modes)}
+          {:ok, div(size, bytes) + remainder}
```

Comment:
Currently we are subtracting the number of BOM bytes from the enumerable size, which doesn't seem correct. Since these bytes are skipped, we should subtract them from the size we consider. I also added a corresponding test (the current one uses `bytes = 1`, which made the calculation equivalent).